qemu/tests/virtio_driver_sign_check.py

```diff
@@ -1,87 +1,153 @@
+        driver_dir = match_ret.group()
+
+        driver_sys = filter(lambda x: driver_dir in x, drivers['.sys'])
+        driver_inf = filter(lambda x: driver_dir in x, drivers['.inf'])
+        driver_wdf = filter(lambda x: driver_dir in x, drivers['Wdf'])
+
+        cat_sys = map(lambda x: (filterd_cat_file, x), driver_sys)
+        cat_inf = map(lambda x: (filterd_cat_file, x), driver_inf)
+        cat_wdf = map(lambda x: (filterd_cat_file, x), driver_wdf)
+        cat_parm = cat_sys + cat_inf + cat_wdf
+        virtio_drivers.extend(cat_parm)
+
     try:
-        for drive in drive_list.split():
-            for type in ['.cat', '.sys']:
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        disk_result = session.cmd_output(get_disk_letter)
+        disk_letter = re.search(r'(\w:)\s', disk_result, re.M).group(1)
+        drive_volume_list.append(disk_letter)
+
+        for drive in drive_volume_list:
+            for type in ['.cat', '.sys', '.inf', 'Wdf']:
                 driver = session.cmd_output(list_files_cmd % (drive, type)).\
                     splitlines()[1:-1]
                 drivers[type] = driver
-            files = zip(drivers['.cat'], drivers['.sys'])
-            for cat, sys in files:
-                cmd = signtool_cmd % (cat, sys)
-                s, result = session.cmd_status_output(cmd)
-                if s:
-                    msg = "Fail command: %s. Output: %s" % (cmd, result)
-                    raise error.TestFail(msg)
+
+            cat_list = filter(filter_all_tested_cat_files, drivers['.cat'])
+            map(match_cat_and_verified_file, cat_list)
+
+            for driver_file in virtio_drivers:
+                test_cmd = signtool_cmd % (driver_file[0], driver_file[1])
+                test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    test.fail(msg)
                 results_all += result
                 re_suc = "Number of files successfully Verified: ([0-9]*)"
                 try:
-                    suc_num = re.findall(re_suc, result)[0]
+                    suc_num = re.search(re_suc, result, re.M).group(1)
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    test.fail(msg)
 
                 if int(suc_num) != 1:
                     fails_log += result
                     fail_num += 1
-                    fail_driver = cat + " " + sys
-                    fail_drivers.append(fail_driver)
+                    fail_drivers.append(driver_file[1])
+            virtio_drivers = []
+
         if fail_num > 0:
             msg = "Following %s driver(s) signature checked failed." % fail_num
```

Comment:
once a signtool check fail, the whole job will be interrupted, so fail_num value can be only be 0 or 1.
1.better to use if fail_num;
2.msg:   driver(s) ---> driver
3.line 143 to line 147 seems useless as line 126 to 128 has detected the error.

Reply:
Hi lijin, 
Look at 125 line: "status, result = session.cmd_status_output(test_cmd)", 
we consider this situation:  the cmd completely(status return 0) and the the result include : Number of files successfully Verified: 0 (I hit this info before).
1. due to the status return 0, so the 126-128 line will not be execute; 
2. then we look 132 line, the suc_num will equal to 0, then we see 137 -139 lines, "if int(suc_num) != 1" equal true, the fail_num=1 due to execute the "fail_num += 1", 
3. next, the code will point to 122 line due to in a loop here, will execute to verify next file, when execute 132 line, also may hit the upper situation, if hit, the fail_num=2. it will verify the file one by one in the loop.
4. after all files be verified, 143-147 lines start be executed,  "if fail_num > 0:" will print "msg = "Following %s driver(s) signature checked failed." % fail_num......" if hit upper situation these codes will be executed.
5. The whole job will be interrupted  when hit these 2 situations:
1). when the cmd status !=0, this time mostly due to the tool file cannot be found or the cmd timeout, will trigger 128 line test.fail(msg).
2).  the cmd result don't have "Number of files successfully Verified: ([0-9]*)" this info, will trigger 135 line test.fail(msg).

Reply:
Could you check my previous comment? If number of successfully verifed is 0, the status is not 0

Reply:
Hi lijin, I tried similar test on my host, I understood wrong the status return value for def cmd_status_output, it's return 0 only the cmd run without any error, not execute completely.

For this patch update:
1. I need the file is verified one by one, so it cannot be interrupt by someone fail,  on 122 line, I cancel interrupt trigger, just show failed cmd and output info.  Totally failed result will use the last 143-147 lines code to trigger. It can be make sure all test can be finished.
2. Changed the disk_letter get way, thanks for you advice.
3. 133 -135 lines need to keep, it will re define the exception  IndexError of re.findall(). It's use to explain this error. 

Please help to review it again, thanks a lot~

PASS 1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.virtio_driver_sign_check